README.md

```diff
@@ -27,3 +27,13 @@ about the current recursive call.
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. You do not need to prove that the invariant is correct. Add
 your answer to this markdown file.
+
+I had the help of chat gpt with the conversion of making my program recursive instead of direct.
+
+A good invariant for recursive implemntation of the fib function would be to get the first few base cases being when n is 0 and when n is 1 ( you can also add if n is negative but thats more for just catching it before it happens). The first base cases you would have a good fibonacci sequence being [0] for 0 and [0,1] for 1.
```

Comment:
An invariant tells you what is true before the loop runs. What would that be here?

Reply:
It would show that the start of the fibonacci sequence is correct

Reply:
And what about the current iteration? The invariant needs to say something about that.

Reply:
during the current iteration the invariant will say that everything n-1 of the current iteration is correctly implemented for the fib sequence

Reply:
...and some version of that should be your invariant.

Reply:
isnt that what the if statements are doing though at the beginning of the code

Reply:
Your invariant is independent of the code.